### PR TITLE
Clean up DpiHelper

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This change cleans up the DpiHelper by clarifying terminology, making behavior more consistent, and fixing a number of issues.

- Rename DpiHelper to ScaleHelper
- Use OS terms
- Use more descriptive naming
- Make scaling actions a no-op if scaling to 100% and just call ScaleHelper by default
- Encapsulate logical default values to avoid accidental unscaled use
- Keep parking window tracking within Application
- Remove System.Window.Forms dependency on SystemEvents
- Remove unreachable code from VsPropertyGrid
- Fixes the Application default font scaling.
- Update some comments (MonthCalendar comments were still in Java)

Most controls that scale use the initial System DPI and don't do any further scaling.

The idea here is that being explicit about what we're scaling to makes things clearer. Always using the ScaleHelper, reducing the caching of scaled values, and factoring the scaling together will make understanding the current behavior easier and make implementing dynamic scaling easier.

DockEditor.DockUI is an example where understanding class state is significantly easier.

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/DpiHelper.cs [hunk 1]
```diff
@@ -3,92 +3,110 @@
...
 using Microsoft.Win32;
 
 namespace System.Windows.Forms;
 
 /// <summary>
-///  Helper class for scaling coordinates and images according to current DPI scaling set in Windows for the primary screen.
+///  Helper class for scaling.
 /// </summary>
-internal static partial class DpiHelper
+internal static partial class ScaleHelper
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/DpiHelper.cs:14]: Which this does, so I've changed it.

File: src/System.Windows.Forms.Design/src/System/Drawing/BitmapSelector.cs [hunk 2]
```diff
@@ -1,79 +1,45 @@
...
 using System.Drawing.Configuration;
 using System.Reflection;
 
 namespace System.Drawing;
 
 /// <summary>
-/// Provides methods to select from multiple bitmaps depending on a "bitmapSuffix" config setting.
+///  Provides methods to select from multiple bitmaps depending on a "bitmapSuffix" config setting.
 /// </summary>
 internal static class BitmapSelector
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Drawing/BitmapSelector.cs:13]: In theory others could utilize this as well. I hope they aren't, but it isn't much to carry.

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/DpiHelper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/DpiHelper.cs:14]: > And I'll update the file name in another change to make sure I don't mess up history.

You can do a "fixup" commit making the requested change followed by an interactive rebase which will amalgamate the fixup into the original commit.

**Review by lonitra — APPROVED** [review]
LGTM. Nice diff!

**Comment by DJm00n** [comment]
@JeremyKuhne maybe you can change `96` here to the `PInvoke.USER_DEFAULT_SCREEN_DPI` const?

https://github.com/dotnet/winforms/blob/e199c86138cf8fee176bddd3f7e19ec62a526765/src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/ScaleHelper.cs#L25

**Pull request merged**